pytest_loki/__init__.py

```diff
@@ -178,15 +192,18 @@ def _push_to_loki(self, streams: List[Tuple[Dict, List[str]]]) -> str:
             }
             payload['streams'].append(stream)
 
-        error_msg = None
-        try:
-            r = requests.post(self._loki_url, json=payload, auth=self._basic_auth)
-        except requests.RequestException as e:
-            error_msg = f"Could not send tests results to loki: {e}"
-        else:
-            if not r.ok:
-                error_msg = f"Failed to send tests resulsts to loki: {r.status_code}: {r.content}"
-        return error_msg
+        for i in range(self._retry_attempts):
+            try:
+                r = requests.post(self._loki_url, json=payload, auth=self._basic_auth)
+                r.raise_for_status()  # This will raise an HTTPError if the request returned a status error
+                return  # If no exception was raised, the request was successful, return from the function
+            except requests.RequestException as e:
```

Comment:
Should we switch to a catch-all try ? We've seen that requests can end up raising an urllib exception that does not fall under the `requests.RequestException` class